src/transformers/models/bert/modeling_bert.py

```diff
@@ -884,6 +1004,9 @@ def __init__(self, config, add_pooling_layer=True):
 
         self.pooler = BertPooler(config) if add_pooling_layer else None
 
+        self._use_sdpa = config._attn_implementation == "sdpa"
```

Comment:
Alright but we can just check directly, this does not scale well for FlashAttention with `_use_flash_attention_2` 😉

Reply:
`self._use_sdpa` check is cheaper than `config._attn_implementation == "sdpa"`

Reply:
IMHO they are basically the same, and just a stylistic choice. Let me know which one you want and I'd be happy to change it.


For context, I looked around a bunch of other models for reference, and they tend to do this for FA2 and sdpa (example below is from `modeling_llama`):

```
        self._use_sdpa = config._attn_implementation == "sdpa"
        self._use_flash_attention_2 = config._attn_implementation == "flash_attention_2"
```

Reply:
we can just store `self.att_implementation = config._attn_implementation`.

Reply:
> For context, I looked around a bunch of other models for reference, and they tend to do this for FA2 and sdpa (example below is from modeling_llama):

yep, and it is not good 😉

Reply:
Gotcha, updated.